src/tools/crossgen2/Common/Compiler/TypeExtensions.cs

```diff
@@ -72,7 +72,7 @@ public static bool RequiresInstMethodDescArg(this MethodDesc method)
         /// </summary>
         public static bool RequiresInstMethodTableArg(this MethodDesc method)
         {
-            return (method.Signature.IsStatic || method.ImplementationType.IsValueType) &&
+            return (method.Signature.IsStatic || method.ImplementationType.IsValueType || (method.OwningType.IsInterface && !method.IsAbstract)) &&
                 method.IsSharedByGenericInstantiations &&
                 !method.HasInstantiation;
         }
```

Comment:
Can you also update `AcquiresInstMethodTableFromThis` below? These three methods need to agree with each other.

Reply:
Done